changelogs/unreleased/gh-7901-iproto-request-handlers-overriding.md

```diff
@@ -0,0 +1,4 @@
+## feature/box
+
```

Comment:
We can't replace the hash table with an array, because the user can override arbitrary request codes, not necessarily existing ones. We considered using an array *in addition to* the hash table as a further optimization, but I think we can do without it, because in my understanding a hash table is almost as fast as a plain array in case it doesn't have a lot of entries. @alyapunov is that right?

Reply:
Yes, I brought this question up, but the problem with using an array is that we want the user to be able to override any request type (i.e., any value ranging from `0` to `2^32`.

Reply:
Allowance to override anything doesn't looks good. We won't be able to safely add new request types ourselves - they would potentially clash with users who introduced their own request types. Would be better to introduce a new request like like IPROTO_NOP which users would override and add their own protocol on top of it. Or give them a safe range to use. But whatever, I won't insist on anything.

Reply:
Ability to override non-existent request types was explicitly asked by the PM (@unera). The rationale is that this would allow one to implement functionality from newer versions. For example, if we had `box.iproto.override` in 2.8, we could implement a Lua module that would add support of IPROTO watchers from 2.10.